main.go

```diff
@@ -9,43 +9,79 @@
 package testcli
 
 import (
+	"bufio"
 	"bytes"
 	"errors"
 	"io"
-	"log"
 	"os"
 	"os/exec"
 	"regexp"
 	"strings"
+	"sync"
+	"testing"
+	"time"
 )
 
+type output struct {
+	content string
+	mu      *sync.Mutex
+}
+
 // Cmd is typically constructed through the Command() call and provides state
 // to the execution engine.
 type Cmd struct {
 	cmd       *exec.Cmd
 	env       []string
 	exitError error
-	executed  bool
-	stdout    string
-	stderr    string
+	status    string
+	stdout    *output
+	stderr    *output
 	stdin     io.Reader
+	t         *testing.T
 }
 
 // ErrUninitializedCmd is returned when members are accessed before a run, that
 // can only be used after a command has been run.
 var ErrUninitializedCmd = errors.New("You need to run this command first")
-var pkgCmd = &Cmd{}
+
+// ErrCmdNotFinished is returned when members are accessed before or during a run,
+// that can only be used after a command has finished executing.
+var ErrCmdNotFinished = errors.New("Command is still executing")
+
+const (
+	// INITIALIZED represents the state of Command before it's started with Run() or Start()
+	INITIALIZED = "initialized"
+	// RUNNING represents the state of Command while it's running
+	RUNNING = "running"
+	// FINISHED represents the state of Command after it has exited successfully or not
+	FINISHED = "finished"
```

Comment:
Please use Camel Case notation for variables and constants. See https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps.

Reply:
Ah, didn't know golang had different standard for constant naming, will get that updated!